lib/packwerk/commands/base_command.rb

```diff
@@ -0,0 +1,49 @@
+# typed: strict
+# frozen_string_literal: true
+
+module Packwerk
+  module Commands
+    class BaseCommand
+      extend T::Sig
+      extend T::Helpers
+      abstract!
+
+      @description = T.let("", String)
+
+      class << self
+        extend T::Sig
+
+        sig { params(description: T.nilable(String)).returns(String) }
+        def description(description = nil)
+          if description
+            @description = description
+          else
+            @description
+          end
+        end
+      end
+
+      sig do
+        params(
+          args: T::Array[String],
+          configuration: Configuration,
+          out: T.any(StringIO, IO),
+          err_out: T.any(StringIO, IO),
+          progress_formatter: Formatters::ProgressFormatter,
+          offenses_formatter: OffensesFormatter,
+        ).void
+      end
+      def initialize(args, configuration:, out:, err_out:, progress_formatter:, offenses_formatter:)
+        @args = args
+        @configuration = configuration
+        @out = out
+        @err_out = err_out
+        @progress_formatter = progress_formatter
+        @offenses_formatter = offenses_formatter
```

Comment:
I'm interested in how you feel about using ivars for stdout/stderr/args/formatters? Should we create private/protected/public readers for these ivars? It seems like it would be a nicer public API to me.

Reply:
Yeah, I like the idea. I tend to define accessors for everything usually. I did not do it here because Sorbet requires a level of verbosity I was not super comfortable with. I'll give it a try.

Reply:
2d9c5c02d34ba7f90df9c65a2f4bf21637e70973